### PR TITLE
Networks: Adding Network Support

### PR DESCRIPTION
**Pull request by eric-ships**
I added a section for what it would take to add network support.

Screenshot 2024-12-17 at 2 58 28 PM

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: docs/pages/features/networks.mdx [hunk 1]
```diff
@@ -34,4 +34,14 @@ Users have the same Smart Wallet address across supported networks.
...
 - `tx.origin` will be the bundler EOA address, not the Smart Wallet address
-- Consider this when integrating with existing contracts
\ No newline at end of file
+- Consider this when integrating with existing contracts
+
+## Adding Network Support
+
+For a network to be supported, it needs:
+
+- CREATE2 deployer at `0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7`
```
Comment (ajhodges) [inline docs/pages/features/networks.mdx:43]:
```suggestion
- [Safe Singleton Factory](https://github.com/safe-global/safe-singleton-factory?tab=readme-ov-file#how-to-get-the-singleton-deployed-to-your-network) at `0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7`
```

File: docs/pages/features/networks.mdx [hunk 2]
```diff
@@ -34,4 +34,14 @@ Users have the same Smart Wallet address across supported networks.
...
+- ERC-4337 Entrypoint v0.6 at `0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789`
```
Comment (ajhodges) [inline docs/pages/features/networks.mdx:44]:
```suggestion
- [ERC-4337 Entrypoint v0.6](https://github.com/eth-infinitism/account-abstraction/releases/tag/v0.6.0) at `0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789` 
```

File: docs/pages/features/networks.mdx [hunk 3]
```diff
@@ -34,4 +34,14 @@ Users have the same Smart Wallet address across supported networks.
...
\ No newline at end of file
+- Consider this when integrating with existing contracts
+
+## Adding Network Support
+
+For a network to be supported, it needs:
+
+- CREATE2 deployer at `0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7`
+- ERC-4337 Entrypoint v0.6 at `0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789` 
+- Smart Wallet Factory at `0x0BA5ED0c6AA8c49038F819E587E2633c4A9F428a`
```
Comment (ajhodges) [inline docs/pages/features/networks.mdx:45]:
```suggestion
- [Smart Wallet Factory](https://github.com/coinbase/smart-wallet/tree/main?tab=readme-ov-file#deployments) at `0x0BA5ED0c6AA8c49038F819E587E2633c4A9F428a`
```

**Review by ajhodges — CHANGES_REQUESTED** [review]
It'd be nice to include some pointers I think

**Comment by cb-heimdall** [comment]
Review Error for ajhodges @ 2024-12-17 15:37:40 UTC
User must have write permissions to review

File: docs/pages/features/networks.mdx [hunk 4]
```diff
@@ -34,4 +34,14 @@ Users have the same Smart Wallet address across supported networks.
...
+- Consider this when integrating with existing contracts
+
+## Adding Network Support
+
+For a network to be supported, it needs:
+
+- [Safe Singleton Factory](https://github.com/safe-global/safe-singleton-factory?tab=readme-ov-file#how-to-get-the-singleton-deployed-to-your-network) (CREATE2 deployer) deployed at `0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7`
+- ERC-4337 Entrypoint v0.6 deployed at `0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789` 
+- Smart Wallet Factory deployed at `0x0BA5ED0c6AA8c49038F819E587E2633c4A9F428a`
+- ERC-4337 bundler meeting Coinbase's standards (available, latency, security)
```
Comment (amiecorso) [inline docs/pages/features/networks.mdx:46]:
```suggestion
- ERC-4337 bundler meeting Coinbase's standards (liveness, latency, security)
```

The use of one adjective and two nouns was throwing me off here.  Would suggest a term like "liveness" or perhaps "availability", though that didn't feel particularly meaningful.

**Review by amiecorso — DISMISSED** [review]
just one comment. thanks for all these docs updates this is great!

File: docs/pages/features/networks.mdx (hunk 4 above)
Comment (eric-ships) [reply docs/pages/features/networks.mdx:46]: Nice catch. This should be [availability](https://en.wikipedia.org/wiki/Reliability,_availability_and_serviceability), given its an eng term

Comment (eric-ships) [reply docs/pages/features/networks.mdx]: Fixed [availability](https://github.com/coinbase/smart-wallet-docs/pull/110/commits/f519c85ba05f1baa869ec664165fd8b83f665573)

File: docs/pages/features/networks.mdx (hunk 2 above)
Comment (eric-ships) [reply docs/pages/features/networks.mdx]: Fixed [657f300](https://github.com/coinbase/smart-wallet-docs/pull/110/commits/657f300186946b550ea57c827c589513cf8f846d)

**Review by ajhodges — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for ajhodges @ 2024-12-17 23:06:07 UTC
User must have write permissions to review

**Review by ajhodges-cb — APPROVED** [review]

**Pull request merged**